cdap-master/src/main/java/co/cask/cdap/data/tools/ExistingEntitySystemMetadataWriter.java

```diff
@@ -142,14 +148,21 @@ private void writeSystemMetadataForDatasets(NamespaceId namespace, DatasetFramew
     SystemDatasetInstantiatorFactory systemDatasetInstantiatorFactory =
       new SystemDatasetInstantiatorFactory(locationFactory, dsFramework, cConf);
     try (SystemDatasetInstantiator systemDatasetInstantiator = systemDatasetInstantiatorFactory.create()) {
+      UserGroupInformation ugi = impersonator.getUGI(namespace.toId().toEntityId());
```

Comment:
@anwar6953 `namespace.toId()` provides Id.Namespace and `namespace.toId().toEntityId()` provides NamesapceId which is needed for `impersonator.getUGI()`